### PR TITLE
google docstring conversion batch (first ~5 directories)

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/10732?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10732/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10732
```

</p>
</details>

### What changes are proposed in this pull request?
Convert the following directories to google format
* mlflow/artifacts
* mlflow/azure
* mlflow/catboost
* mlflow/data
* mlflow/* (non-recursive)
### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/__init__.py [hunk 1]
```diff
@@ -121,7 +121,6 @@
     delete_run,
     delete_tag,
     end_run,
-    flush_async_logging,
     get_artifact_uri,
```
Comment (michael-berk) [inline mlflow/__init__.py:124]: BERK: bad drop

File: mlflow/artifacts/__init__.py [hunk 2]
```diff
@@ -134,29 +140,13 @@ def load_dict(artifact_uri: str) -> dict:
...
-            image = Image.new("RGB", (100, 100))
-            artifact_uri = run.info.artifact_uri
-            mlflow.log_image(image, "image.png")
-            image = mlflow.artifacts.load_image(artifact_uri + "/image.png")
-            print(image)
-
-    .. code-block:: text
-        :caption: Output
+    Args:
+        artifact_uri: Artifact location.
```
Comment (michael-berk) [inline mlflow/artifacts/__init__.py:146]: BERK: bad drop

File: mlflow/artifacts/__init__.py [hunk 3]
```diff
@@ -102,27 +106,29 @@ def load_text(artifact_uri: str) -> str:
...
+        artifact_uri: artifact location.
 
-    :param artifact_uri: artifact location
-    :return: dict
+    Returns:
+        A dictionary.
 
     .. code-block:: python
-        :caption: Example
+      :caption: Example
```
Comment (michael-berk) [inline mlflow/artifacts/__init__.py:118]: BERK: TODO check if 2 space indent builds

File: mlflow/diviner/__init__.py [hunk 4]
```diff
@@ -102,52 +104,52 @@ def save_model(
...
+            the training dataset), for example:
+
+            .. code-block:: python
+
+                from mlflow.models import infer_signature
+
+                model = diviner.GroupedProphet().fit(data, ("region", "state"))
+                predictions = model.predict(prediction_config)
+                signature = infer_signature(data, predictions)
+
```
Comment (michael-berk) [inline mlflow/diviner/__init__.py:132]: Removed parameterizations

File: mlflow/diviner/__init__.py [hunk 5]
```diff
@@ -102,52 +104,52 @@ def save_model(
...
-                   - `dfs_tmpdir` for specifying the DFS temporary location where the model will \
-                   be stored while copying from a local file system to a Spark-supported "dbfs:/" \
-                   scheme.
+    """Save a ``Diviner`` model object to a path on the local file system.
+
+    Args:
+        diviner_model: ``Diviner`` model that has been ``fit`` on a grouped temporal
+            ``DataFrame``.
+        path: Local path destination for the serialized model is to be saved.
+        conda_env: Inserted after the argument name.
```
Comment (michael-berk) [inline mlflow/diviner/__init__.py:113]: BERK: removed parametrization

File: mlflow/data/numpy_dataset.py [hunk 6]
```diff
@@ -159,50 +162,44 @@ def from_numpy(
...
+            the source is assumed to be the code location (e.g. notebook cell, script, etc.) where
+            from_numpy is being called.
+        targets: Optional NumPy targets, represented as an np.ndarray or dictionary of named
+            np.ndarrays.
+        name: The name of the dataset. If unspecified, a name is generated.
+        digest: The dataset digest (hash). If unspecified, a digest is computed automatically.
 
     .. testcode:: python
-        :caption: Basic Example
+      :caption: Basic Example
```
Comment (michael-berk) [inline mlflow/data/numpy_dataset.py:182]: BERK: check this compiles

File: mlflow/artifacts/__init__.py [hunk 7]
```diff
@@ -70,11 +72,13 @@ def download_artifacts(
...
-    Loads the artifact contents as a string.
+    """Loads the artifact contents as a string.
+
+    Args:
+        artifact_uri: Artifact location.
 
-    :param artifact_uri: artifact location
-    :return: str
+    Returns:
+        The contents of the artifact as a string.
```
Comment (michael-berk) [inline mlflow/artifacts/__init__.py:81]: This is a reasonable improvement.

**Comment by michael-berk** [comment]
Here are all the diffs in the render html. They seem fine
**Good - formatting change with no changes to rendered docs.**
```
mlflow.diviner.html
------------------------------
855: +                model instance.
1385: +                     Example
1692: +                  – Additional arguments for
1702: +                    mlflow.models.model.Model
1711: +                  Additionally, for models that have been fit in Spark, the following supported
1714: + -
1724: +                   partition_by
1730: +                  for setting a (or several) partition columns as a list of             column names. Must be a list of strings of grouping key column(s).
1731: + -
1741: +                   partition_count
1747: +                  for setting the number of part files to write from a             repartition per
1753: +                   partition_by
1759: +                  group. The default part file count is 200.
1760: + -
1770: +                   dfs_tmpdir
1776: +                  for specifying the DFS temporary location where the model will             be stored while copying from a local file system to a Spark-supported “dbfs:/”             scheme.
2576: +                  – Optional configurations for Spark DataFrame storage iff the model has
2583: + Current supported options:
2584: + -
2594: +                   partition_by
2600: +                  for setting a (or several) partition columns as a list of             column names. Must be a list of strings of grouping key column(s).
2601: + -
2611: +                   partition_count
2617: +                  for setting the number of part files to write from a             repartition per
2623: +                   partition_by
2629: +                  group. The default part file count is 200.
2630: + -
2640: +                   dfs_tmpdir
2646: +                  for specifying the DFS temporary location where the model will             be stored while copying from a local file system to a Spark-supported “dbfs:/”             scheme.
854: -                model instance
1689: -                  –
1691: -                   Additional arguments for
1700: -                     mlflow.models.model.Model
1709: -                   Additionally, for models that have been fit in Spark, the following supported
1722: -                      partition_by
1728: -                     for setting a (or several) partition columns as a list of                    column names. Must be a list of strings of grouping key column(s).
1739: -                      partition_count
1745: -                     for setting the number of part files to write from a                    repartition per
1751: -                      partition_by
1757: -                     group. The default part file count is 200.
1768: -                      dfs_tmpdir
1774: -                     for specifying the DFS temporary location where the model will                    be stored while copying from a local file system to a Spark-supported “dbfs:/”                    scheme.
2573: -                  –
2575: -                   Optional configurations for Spark DataFrame storage iff the model has
2581: -                   Current supported options:
2592: -                      partition_by
2598: -                     for setting a (or several) partition columns as a list of                    column names. Must be a list of strings of grouping key column(s).
2609: -                      partition_count
2615: -                     for setting the number of part files to write from a                    repartition per
2621: -                      partition_by
2627: -                     group. The default part file count is 200.
2638: -                      dfs_tmpdir
2644: -                     for specifying the DFS temporary location where the model will                    be stored while copying from a local file system to a Spark-supported “dbfs:/”                    scheme.
```

**Good - formatting and capitalization change.**
```
mlflow.artifacts.html
------------------------------
904: +                – artifact location.
914: +                A dictionary.
1023: +                – Artifact location.
1033: +                A PIL.Image object.
1142: +                – Artifact location.
1152: +                The contents of the artifact as a string.
903: -                – artifact location
913: -                dict
1021: -                – artifact location
1032: -                PIL.Image
1140: -                – artifact location
1151: -                str
```

**Good - line wrap changes**
```
mlflow.data.html
------------------------------
1390: +                 Args:
1405: +                   dataset:
1415: +                    An instance of
1427: +                       mlflow.data.dataset.Dataset
1439: +                    ,
1451: +                       mlflow.entities.Dataset
1463: +                    , or
1475: +                       mlflow.entities.DatasetInput
1487: +                    .
1500: +                 Returns:
1511: +                  An instance of
1523: +                     DatasetSource
1535: +                  .
2534: +                   – NumPy features, represented as an np.ndarray or dictionary of named np.ndarrays.
2545: +                   – The source from which the numpy data was derived, e.g. a filesystem path, an S3 URI,
2547: + an HTTPS URL, a delta table name with version, or spark table etc.
2557: +                   may be
2558: + specified as a URI, a path-like string, or an instance of
2566: +                   . If unspecified,
2567: + the source is assumed to be the code location (e.g. notebook cell, script, etc.) where
2604: +                   – The dataset digest (hash). If unspecified, a digest is computed automatically.
3604: +                   – The path of the Spark or Delta source that the DataFrame originally came from. Note
3608: + that the path does not have to match the DataFrame exactly, since the DataFrame may have
3610: + been modified by Spark operations. This is used to reload the dataset upon request via
3639: +                   are specified, a CodeDatasetSource is used, which will source
3643: + information from the run context.
3653: + Note that the table does not have to match the DataFrame exactly, since the DataFrame
3657: + may have been modified by Spark operations. This is used to reload the dataset upon
3660: + request via
3687: +                   are specified, a CodeDatasetSource is used, which will source
3690: + information from the run context.
3699: +                   – If the DataFrame originally came from a Delta table, specifies the version of the
3703: + Delta table. This is used to reload the dataset upon request via
3719: +                   cannot be
3720: + specified if
3735: +                   – The Spark SQL statement that was originally used to construct the DataFrame. Note that
3739: + the Spark SQL statement does not have to match the DataFrame exactly, since the
3743: + DataFrame may have been modified by Spark operations. This is used to reload the dataset
3746: + upon request via
3773: +                   are specified, a CodeDatasetSource is used, which will source
3777: + information from the run context.
3795: +                   – The name of the dataset. E.g. “wiki_train”. If unspecified, a name is automatically
3798: + generated.
3807: +                   – The digest (hash, fingerprint) of the dataset. If unspecified, a digest is
3811: + automatically computed.
5089: +               If the source is path like, then this will construct a DatasetSource object from the source
6221: +                  Returns
6225: +                   An instance of
6228: +                     pyspark.sql.DataFrame
6231: +                   .
6344: +                  Returns
6348: +                   An instance of
6351: +                     pyspark.sql.DataFrame
6354: +                   .
1388: -               Parameters
1404: -                 dataset
1413: -                – An instance of
1426: -                   mlflow.data.dataset.Dataset
1438: -                ,
1450: -                   mlflow.entities.Dataset
1462: -                , or
1474: -                   mlflow.entities.DatasetInput
1486: -                .
1499: -               Returns
1510: -                An instance of
1522: -                   DatasetSource
1534: -                .
2533: -                   – NumPy features, represented as an np.ndarray or dictionary of named
2536: - np.ndarrays.
2544: -                   – The source from which the numpy data was derived, e.g. a filesystem
2548: - path, an S3 URI, an HTTPS URL, a delta table name with version, or
2549: - spark table etc.
2555: -                   may be specified as a URI, a path-like string,
2556: - or an instance of
2568: -                   .
2569: - If unspecified, the source is assumed to be the code location
2570: - (e.g. notebook cell, script, etc.) where
2603: -                   – The dataset digest (hash). If unspecified, a digest is computed
2606: - automatically.
3603: -                   – The path of the Spark or Delta source that the DataFrame originally came from.
3606: - Note that the path does not have to match the DataFrame exactly, since the
3611: - DataFrame may have been modified by Spark operations. This is used to reload the
3612: - dataset upon request via
3638: -                   are specified, a CodeDatasetSource is used, which will
3641: - source information from the run context.
3652: - Note that the table does not have to match the DataFrame exactly, since the
3655: - DataFrame may have been modified by Spark operations. This is used to reload
3659: - the dataset upon request via
3686: -                   are specified, a CodeDatasetSource is
3689: - used, which will source information from the run context.
3698: -                   – If the DataFrame originally came from a Delta table, specifies the version
3701: - of the Delta table. This is used to reload the dataset upon request via
3717: -                   cannot be specified if
3734: -                   – The Spark SQL statement that was originally used to construct the DataFrame.
3737: - Note that the Spark SQL statement does not have to match the DataFrame exactly,
3741: - since the DataFrame may have been modified by Spark operations. This is used to
3745: - reload the dataset upon request via
3772: -                   are specified, a CodeDatasetSource is used,
3775: - which will source information from the run context.
3794: -                   – The name of the dataset. E.g. “wiki_train”. If unspecified, a name is
3797: - automatically generated.
3806: -                   – The digest (hash, fingerprint) of the dataset. If unspecified, a digest
3809: - is automatically computed.
4760: -                   instance.
5088: - If the source is path like, then this will construct a DatasetSource object from the source
6211: - :return: An instance of
6214: -                   pyspark.sql.DataFrame
6217: -                 .
6334: - :return: An instance of
6337: -                   pyspark.sql.DataFrame
6340: -                 .
```

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Pull request merged**